### PR TITLE
convert storage to atom for ensemble user

### LINKED ISSUE #386: Allow functions on ensemble user atom
**Is your feature request related to a problem? Please describe.**
We need to be able to add functions on ensemble user atom so we can integrate with third party auth services. Functions cannot be stored in session/local storage and must be retained in memory.

**Describe the solution you'd like**
Update the ensemble user atom to support function values. Attributes can still remain in session storage but the user atom should also be mirrored in the application context

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
to provide the function storage inside the ensemble user, I converted the storage atom to simple atom for ensemble user

## Issue ticket number and link

Closes #386 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/framework/src/hooks/useEnsembleUser.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { useAtom } from "jotai";
+import { atomWithStorage, createJSONStorage } from "jotai/utils";
+import { merge } from "lodash-es";
+import { useMemo } from "react";
+
+export type EnsembleUser = { accessToken?: string } & Record<string, unknown>;
+export type EnsembleUserBuffer = {
```
Comment (evshi) [inline packages/framework/src/hooks/useEnsembleUser.ts:7]: Don't expose this type externally. Just keep it all under EnsembleUser

**Review by evshi — COMMENTED** [review]
Please fix typing and add changeset

File: packages/framework/src/evaluate/binding.ts [hunk 2]
```diff
@@ -92,9 +92,7 @@ export const createBindingAtom = <T = unknown>(
             ? get(screenStorageAtom)
             : undefined,
         ),
-        user: rawJsExpression.includes("ensemble.user")
-          ? get(userAtom)
-          : undefined,
+        user: get(userAtom),
```
Comment (evshi) [inline packages/framework/src/evaluate/binding.ts:95]: Can you revert this change? We don't want to create unnecessary depenencies.

File: packages/framework/src/evaluate/binding.ts [hunk 3]
```diff
@@ -8,11 +8,11 @@ import {
   createStorageApi,
   screenStorageAtom,
 } from "../hooks/useEnsembleStorage";
+import { userAtom } from "../hooks/useEnsembleUser";
```
Comment (evshi) [inline packages/framework/src/evaluate/binding.ts:11]: Let's keep the userAtom inside the state module. Hook can be its own file.

File: packages/framework/src/evaluate/binding.ts (hunk 2 above)
Comment (sagardspeed2) [reply packages/framework/src/evaluate/binding.ts:95]: so should I remove the get method from the useEmsembleUser hook ?

Comment (evshi) [reply packages/framework/src/evaluate/binding.ts:95]: No, why would you do that? I'm just asking to revert these three lines here.

Comment (sagardspeed2) [reply packages/framework/src/evaluate/binding.ts]: oh, I changed this while I changed user atom from state to hook. this is for the test only and it was need to be reverted back.

**Review by evshi — APPROVED** [review]
